packages/core/src/transport/sendWithRetryStrategy.spec.ts

```diff
@@ -0,0 +1,222 @@
+      sendRequest({ bytesCount: 30 })
+      expect(state.queuedPayloads.size()).toBe(2)
+      expect(state.queuedPayloads.bytesCount).toBe(MAX_QUEUE_BYTES_COUNT + 10)
+
+      sendRequest({ bytesCount: 1 })
+      expect(state.queuedPayloads.size()).toBe(2)
+      expect(state.queuedPayloads.bytesCount).toBe(MAX_QUEUE_BYTES_COUNT + 10)
+    })
+
+    it('should respect request order', () => {
+      sendRequest({ bytesCount: MAX_ONGOING_BYTES_COUNT - 10 })
+      sendRequest({ bytesCount: 20 })
+      sendRequest({ bytesCount: MAX_ONGOING_BYTES_COUNT - 15 })
+      sendRequest({ bytesCount: 10 })
+      expect(state.queuedPayloads.size()).toBe(3)
+      expect(state.queuedPayloads.bytesCount).toBe(20 + (MAX_ONGOING_BYTES_COUNT - 15) + 10)
+
+      sendStub.respondWith(0, { status: 200 })
+      expect(state.queuedPayloads.size()).toBe(2)
+      expect(state.queuedPayloads.bytesCount).toBe(MAX_ONGOING_BYTES_COUNT - 15 + 10)
+
+      sendStub.respondWith(1, { status: 200 })
+      expect(state.queuedPayloads.size()).toBe(0)
+    })
+  })
+  ;[
+    { description: 'when the intake returns error:', status: 500 },
+    { description: 'when network is down:', status: 0 },
+  ].forEach(({ description, status }) => {
+    describe(description, () => {
+      it('should start queueing following requests', () => {
+        sendRequest()
+        sendStub.respondWith(0, { status })
+        expect(state.queuedPayloads.size()).toBe(1)
+
+        sendRequest()
+        expect(state.queuedPayloads.size()).toBe(2)
+        sendRequest()
+        expect(state.queuedPayloads.size()).toBe(3)
+      })
+
+      it('should send queued requests if another ongoing request succeed', () => {
+        sendRequest()
+        sendRequest()
+        sendStub.respondWith(0, { status })
+        expect(state.bandwidthMonitor.ongoingRequestCount).toBe(1)
+        expect(state.queuedPayloads.size()).toBe(1)
+
+        sendRequest()
+        expect(state.bandwidthMonitor.ongoingRequestCount).toBe(1)
+        expect(state.queuedPayloads.size()).toBe(2)
+
+        sendStub.respondWith(1, { status: 200 })
+        expect(state.bandwidthMonitor.ongoingRequestCount).toBe(2)
+        expect(state.queuedPayloads.size()).toBe(0)
+      })
+    })
+  })
+
+  describe('when transport down:', () => {
+    it('should regularly try to send first queued request', () => {
+      sendRequest()
+      sendStub.respondWith(0, { status: 500 })
+      expect(state.bandwidthMonitor.ongoingRequestCount).toBe(0)
+
+      clock.tick(INITIAL_BACKOFF_TIME)
+      expect(state.bandwidthMonitor.ongoingRequestCount).toBe(1)
+      sendStub.respondWith(1, { status: 500 })
+
+      clock.tick(2 * INITIAL_BACKOFF_TIME)
+      expect(state.bandwidthMonitor.ongoingRequestCount).toBe(1)
+      sendStub.respondWith(2, { status: 500 })
+
+      clock.tick(4 * INITIAL_BACKOFF_TIME)
+      expect(state.bandwidthMonitor.ongoingRequestCount).toBe(1)
+      sendStub.respondWith(3, { status: 500 })
```

Comment:
<!-- Nitpicks are small, trivial, but necessary changes. Distinguishing nitpick comments significantly helps direct the reader's attention to comments requiring more involvement.  -->
**🥜 nitpick:** Responding to the request does not look useful if we don't assert anything after that